tests/test_expectations.py

```diff
@@ -101,6 +106,9 @@ def dirac_markov_gauss():
 def rbf_kern():
     return kernels.RBF(Data.D_in, variance=rng.rand(), lengthscales=rng.rand() + 1.)
 
+@cache_tensor
+def rbf_kern_2():
```

Comment:
knowing what this PR is about I understand why you've got essentially the same code twice, but it'd be good to have a short comment explaining this duplication!

Reply:
@sanket-kamthe agreed, add docstring to this test describing what's going on